rigex-front-end/src/components/WellModal/WellModal.js

```diff
@@ -0,0 +1,201 @@
+import React, { useState, useEffect } from 'react';
+import { useMutation } from '@apollo/client';
+import { useDispatch } from 'react-redux';
+import Dialog from '@material-ui/core/Dialog';
+import CloseIcon from '@material-ui/icons/Close';
+
+import { wellActions } from '../../store/wells/action';
+import { UPSERT_WELL } from '../../constants/serviceAPI';
+import './wellModalStyles.scss';
+
+export default function WellModal({
+  wellModalOpen: {
+    type,
+    isOpen,
+    currentName,
+    currentLatitude,
+    currentLongitude,
+    id,
+  },
+  handleWellModalClose,
+  rigId,
+}) {
+  const [wellInfo, setWellInfo] = useState({
+    name: '',
+    latitude: '',
+    longitude: '',
+  });
+  const [isValidInput, setIsValidInput] = useState({
+    name: false,
+    latitude: false,
+    longitude: false,
+  });
+  const [isCreateType, setIsCreateType] = useState(true);
+  const [isActiveSubmit, setIsActiveSubmit] = useState(false);
+  const [upsertWell, { loading, error, data }] = useMutation(UPSERT_WELL);
+  const dispatch = useDispatch();
+
+  const handleSubmit = async (event) => {
+    event.preventDefault();
+    const wellInput = {
+      name: wellInfo.name,
+      longitude: parseFloat(wellInfo.longitude),
+      latitude: parseFloat(wellInfo.latitude),
+      rigId: rigId.toString(),
+    };
+
+    if (id) {
+      wellInput.id = id;
+    }
+
+    await upsertWell({
+      variables: {
+        wellInput: { ...wellInput },
+      },
+    });
+
+    if (event.target.name === 'create') {
+      handleWellModalClose();
+    }
+  };
+
+  useEffect(() => {
+    if (data) {
+      dispatch(wellActions.upsertWell(data.upsertWell));
+    }
+  }, [dispatch, loading, error, data]);
+
+  useEffect(() => {
+    let isValid = true;
+    Object.entries(isValidInput).forEach(([key, value]) => {
+      if (!value) {
+        isValid = false;
+      }
+    });
+    if (isActiveSubmit !== isValid) {
+      setIsActiveSubmit(isValid);
+    }
+  }, [isValidInput, isActiveSubmit]);
+
+  useEffect(() => {
+    setIsCreateType(type === 'create');
+    if (type === 'update') {
+      setWellInfo({
+        name: currentName,
+        latitude: currentLatitude,
+        longitude: currentLongitude,
+      });
+      setIsValidInput({
+        name: true,
+        latitude: true,
+        longitude: true,
+      });
+    }
+  }, [type, isOpen, currentName, currentLatitude, currentLongitude, id]);
+
+  const handleChange = ({ target: { name, value } }) => {
+    let latLongRegExTest = /^-?[0-9]{1,3}(?:\.([0-9]{1,10})?)?$/;
```

Comment:
is there a const file this can live in? that way we can use the same regex expression throughout the app